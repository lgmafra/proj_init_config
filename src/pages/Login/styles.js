import { StyleSheet } from 'react-native';
import { colors, metrics } from '~/styles';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'stretch',
    justifyContent: 'flex-start',
    backgroundColor: colors.darker,
  },
  imageHeader: {
    width: metrics.screenWidth,
    height: 196,
    flex: 1,
  },
  backButton: {
    marginTop: metrics.baseMargin,
  },
  content: {
    width: metrics.screenWidth,
    height: metrics.screenHeight - 162,
    borderTopLeftRadius: metrics.baseRadius * 12,
    borderTopRightRadius: metrics.baseRadius * 12,
    backgroundColor: colors.white,
    padding: metrics.basePadding,
  },
  textInfo: {
    marginTop: metrics.baseMargin * 2,
    fontSize: 28,
    color: colors.secondary,
  },
  textInfoBold: {
    marginTop: metrics.baseMargin * 8,
    fontSize: 28,
    fontWeight: 'bold',
    color: colors.secondary,
  },
  form: {
    marginTop: metrics.baseMargin * 5,
  },
  emailInput: {
    height: 55,
    backgroundColor: colors.inputBackground,
    borderRadius: metrics.baseRadius * 4,
    padding: metrics.basePadding,
    marginBottom: metrics.baseMargin * 2,
  },
  passwordInput: {
    height: 55,
    backgroundColor: colors.inputBackground,
    borderRadius: metrics.baseRadius * 4,
    padding: metrics.basePadding,
  },
  forgotPassword: {
    fontSize: 15,
    color: colors.primary,
    marginTop: metrics.baseMargin,
    marginBottom: metrics.baseMargin * 3,
  },
  buttonLogin: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 136,
    height: 41,
    borderRadius: metrics.baseRadius * 9,
    backgroundColor: colors.primary,
  },
  buttonText: {
    fontSize: 15,
    color: colors.white,
  },
  socialLogos: {
    flexDirection: 'row',
    marginTop: metrics.baseMargin * 4,
  },
  logos: {
    marginRight: metrics.baseMargin * 2,
  },
});

export default styles;
