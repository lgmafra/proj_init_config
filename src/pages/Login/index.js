import React, { useState } from 'react';
import {
  View, Text, ImageBackground, TextInput, TouchableOpacity, Image,
} from 'react-native';

import styles from './styles';

import header from '~/images/header.png';
import google from '~/images/google.png';
import facebook from '~/images/facebook.png';

export default function Login() {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [logged, setLogged] = useState(false);

  function handleLogin() {
    if (email !== '' && password !== '') {
      setLogged(true);
    }
  }

  return (
    <View style={styles.container}>
      <ImageBackground source={header} style={styles.imageHeader} resizeMode="cover" />
      <View style={styles.content}>
        <Text style={styles.textInfo}>
          Fa√ßa
          {' '}
          <Text style={styles.textInfoBold}>login</Text>
        </Text>
        <View style={styles.form}>
          <TextInput
            testID="email"
            style={styles.emailInput}
            placeholder="Email"
            autoCapitalize="none"
            autoCompleteType="email"
            onChangeText={text => setEmail(text)}
            value={email}
          />
          <TextInput
            testID="password"
            style={styles.passwordInput}
            placeholder="Senha"
            autoCapitalize="none"
            autoCompleteType="password"
            onChangeText={text => setPassword(text)}
            value={password}
          />
          <Text style={styles.forgotPassword}>Esqueceu a senha?</Text>
          <TouchableOpacity
            testID="loginButton"
            style={styles.buttonLogin}
            onPress={() => handleLogin}
          >
            <Text style={styles.buttonText}>Login</Text>
          </TouchableOpacity>
        </View>
        <View style={styles.socialLogos}>
          <Image source={facebook} style={styles.logos} />
          <Image source={google} style={styles.logos} />
        </View>
      </View>
    </View>
  );
}
