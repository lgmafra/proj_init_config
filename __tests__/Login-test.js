import React from 'react';
import { shallow } from 'enzyme';
import Login from '../src/pages/Login';

describe('test Login page component', () => {
  const wrapper = shallow(<Login />);

  const loginText = 'sthima@sthima.com';
  const passwordText = 'sthima';

  it('match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  const email = wrapper.findWhere(node => node.prop('testID') === 'email');
  const password = wrapper.findWhere(node => node.prop('testID') === 'password');
  const loginButton = wrapper.findWhere(node => node.prop('testID') === 'loginButton');

  it('login page has 2 inputs', () => {
    expect(email.exists()).toBe(true);
    expect(password.exists()).toBe(true);
  });

  it('login page has login button', () => {
    expect(loginButton.exists()).toBe(true);
  });

  it('can do login', () => {
    email.props().onChangeText(loginText);
    password.props().onChangeText(passwordText);

    loginButton.props().onPress();
  });
});
