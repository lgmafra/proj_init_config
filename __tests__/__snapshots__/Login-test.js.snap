// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Login page component match snapshot 1`] = `ShallowWrapper {}`;
